### PR TITLE
Limiting the scope of variables #874

### PR DESCRIPTION
**Pull request by SalahSoliman**
Limited the scope of variables in moveit_core/collision_detection

### Description
Limited the scope of variables in moveit_core/collision_detection/src/**

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_core/collision_detection/src/collision_octomap_filter.cpp
Comment (AndyZe) [inline moveit_core/collision_detection/src/collision_octomap_filter.cpp:189]: It's strange that this variable doesn't need to be initialized but I checked the implementation of `sampleCloud()` and indeed, it does not need to be initialized.

Let's do it anyway, to be on the safe side.

```suggestion
    double intensity = 0;
```

**Review by AndyZe — COMMENTED** [review]
Thanks! Good idea. I have some nitpicky, minor suggestions.

File: moveit_core/collision_detection/src/collision_octomap_filter.cpp
Comment (SalahSoliman) [reply moveit_core/collision_detection/src/collision_octomap_filter.cpp:212]: I think having a variable is better to self comment what this loop does.

**Review by SalahSoliman — COMMENTED** [review]
Thank you for the suggestions, all of them makes sense

**Review by AndyZe — APPROVED** [review]
Great, we just need to wait for one more review, then we can merge it.

**Review by henningkayser — APPROVED** [review]
Thank you!

**Pull request closed without merging**
